account_payment_partner/views/account_move_line.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!--
+  Copyright 2016 Akretion (http://www.akretion.com/)
+  License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+-->
+<odoo>
+    <record id="view_move_line_form" model="ir.ui.view">
+        <field name="name">account_payment_partner.move_line_form</field>
+        <field name="model">account.move.line</field>
+        <field name="inherit_id" ref="account.view_move_line_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='analytic_distribution']/.." position="after">
+                <group name="payments" string="Payments">
+                    <field name="account_type" invisible="1" />
+                    <field name="reconciled" invisible="1" />
+                    <field
+                        name="payment_mode_id"
+                        force_save="1"
+                        invisible="account_type not in ['asset_receivable', 'liability_payable']"
+                        readonly="account_type not in ['asset_receivable', 'liability_payable'] or reconciled == True"
```

Comment:
And the same for similar cases with booleans:

```suggestion
                        readonly="account_type not in ['asset_receivable', 'liability_payable'] or reconciled"
```

Reply:
Fixed. Thanks.